exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -41,7 +40,9 @@ func (ke kafkaErrors) Error() string {
 }
 
 func (e *kafkaTracesProducer) tracesPusher(_ context.Context, td ptrace.Traces) error {
-	messages, err := e.marshaler.Marshal(td, e.topic)
+	topic := e.cfg.getTopic(td.ResourceSpans().At(0).Resource().Attributes())
```

Comment:
Currently, we are deciding on the topic name based on only the first span, but I believe it's possible to loop through and reference all spans, although there would be a trade-off with performance.

Reply:
I think the the first value for the resource attribute that's seen is acceptable. 

One thing you might consider though, what if spans have different set of resource attributes? It might be good to pass in the whole set of resource spans, and then return the first valid value for the chosen resource attribute.